src/include/storage/block_compactor.h

```diff
@@ -49,6 +54,36 @@ class BlockCompactor {
   };
 
  public:
+  BlockCompactor(execution::exec::ExecutionContext *exec, col_id_t *col_oids, const char *table_name) {
+    // Init data members for movetuple builtin
+    exec_ = exec;
+    col_oids_ = col_oids;
+    table_name_ = table_name;
+    // tpl code for use in moveTuple. It deletes the tuple from the table and from the index and then inserts the tuple
+    // to the table (a specific block) and to the index. It returns true if the delete succeeds (because delete returns
+    // false if a concurrent transaction is updating the tuple that is trying to be moved, the only condition where
+    // the block compaction should be stopped).
+    // This replaces with the index delete/insert added:
+    //      cg->table_->InsertInto(common::ManagedPointer(cg->txn_), *record->Delta(), to);
+    //      return cg->table_->Delete(common::ManagedPointer(cg->txn_), from);
+    // @todo: do we need to consider that insertIndex could fail? Do so now.
+    auto tpl_code = R"(
```

Comment:
Might be better if the tpl code is defined as a constant in the beginning of the file..

Reply:
Is there any better alternative to hardcoding tpl code as a string and compile at runtime?

Reply:
Yes, this needs to be done within CodeGen. But the downside is that we can only do that once the circular dependency with catalog is resolved. We have mentioned this in our design doc.